Dockerfile-dev

```diff
@@ -0,0 +1,26 @@
+FROM tiangolo/uvicorn-gunicorn-fastapi:python3.8-slim
+
+RUN apt-get -y update
+RUN apt-get install -y --fix-missing build-essential cmake libpq-dev curl
+
+RUN pip install 'poetry==1.1.7'
+
+ENV MODULE_NAME="jobbergateapi2.main"
+
+COPY ./pyproject.toml /app/
+COPY ./poetry.lock /app/
+
+ARG PYPI_PASSWORD
+ENV POETRY_HTTP_BASIC_PYPICLOUD_USERNAME=admin
+ENV POETRY_HTTP_BASIC_PYPICLOUD_PASSWORD=$PYPI_PASSWORD
```

Comment:
why not use `hack_build_secrets.sh` here too?

Reply:
Because this Dockerfile-dev is intended to be used for local development (with docker-compose), and secrets will probably not be stored on your workstation in the same way.

To use this Dockerfile-dev by itself, you would do this:
```bash
$ docker build --file=Dockerfile-dev --build-arg="PYPI_PASSWORD=putpasswordhere" .
```

Reply:
I'm still working on setting up the armada-composed `docker-compose.yaml` file, but I will include in the README a description of how to set the build args.